src/main/java/harmonised/pmmo/config/codecs/LocationData.java

```diff
@@ -27,11 +27,12 @@ public record LocationData(
 		Map<ResourceLocation, Integer> negative,
 		List<ResourceLocation> veinBlacklist,
 		Map<String, Integer> travelReq,
+		List<AttributeModifier> dimensionalMobModifiers,
 		Map<ResourceLocation, Map<String, Double>> mobModifiers) implements DataSource<LocationData>{
```

Comment:
Replace the `Map<String, Double>` with `List<MobModifier>` so the same operations can be done on a single mob the same way they would be done globally.

Reply:
So the already existing mobModifiers to be ported to use MobModifiers instead of static KV.
One question though: How should we treat the formula?
1. Fully Cumulative (global biome + global dimension + mob biome + mob dimension)
2. Overriding (mob biome will override global biome if an explicit key for the mob is present + mob dimension will override global dimension if an implicit key for the mob is present)
3. Selective Overriding (like 2, but overriding only existing attributes)

If 2 is the case, then if we have a key for _minecraft:zombie_ in overworld.json, under _mobModifiers_ => We ignore the any changes under _globalMobModifiers_, and apply only _mobModifiers_ for _minecraft:zombie_

If 3 is the case, then if we have a key for _minecraft:zombie_ in overworld.json, under _mobModifiers_ => Under _globalMobModifiers_ we ignore ONLY the attributes that already exist in _mobModifiers_, but keep the attributes that do not match

Reply:
Option 1, and use my suggestion here https://github.com/Caltinor/Project-MMO-2.0/pull/577#discussion_r1681571185 so that getting the dimension and biome modifiers is still only one call.  So it's `dim + biome` but underneath that is actually `(dimGlobal + dimMob) + (biomeGlobal + biomeMob)`.

Reply:
Porting the mobModifiers from LocationData required a fair amount of refactoring all around. Did a refactor on MobAttributeHandler considering the new ability to scale with multiple operation types.